service.subtitles.subspedia/service.py

```diff
@@ -182,59 +205,41 @@ def get_params():
     item['episode']            = str(xbmc.getInfoLabel("VideoPlayer.Episode"))                   # Episode
     item['tvshow']             = normalizeString(xbmc.getInfoLabel("VideoPlayer.TVshowtitle"))   # Show
-    item['title']              = normalizeString(xbmc.getInfoLabel("VideoPlayer.OriginalTitle")) # try to get original title
-    item['file_original_path'] = urllib.unquote(xbmc.Player().getPlayingFile().decode('utf-8'))  # Full path of a playing file
     item['3let_language']      = []
   
     for lang in urllib.unquote(params['languages']).decode('utf-8').split(","):
         item['3let_language'].append(xbmc.convertLanguage(lang,xbmc.ISO_639_2))
     
-    if item['title'] == "":
+    if not item['tvshow']:
         item['title']  = normalizeString(xbmc.getInfoLabel("VideoPlayer.Title"))      # no original title, get just Title
+        toParse = item['title'].lower().replace('.', " ")
+        infoFromTitle = parseSearchString(toParse)
+        if infoFromTitle:
+            item['tvshow'] = infoFromTitle['tvshow']
+            item['season']= infoFromTitle['season']
+            item['episode']= infoFromTitle['episode']
+        else:
+            notify(__language__(32002))
       
     if item['episode'].lower().find("s") > -1:                                      # Check if season is "Special"
         item['season'] = "0"                                                          #
         item['episode'] = item['episode'][-1:]
-    
-    if ( item['file_original_path'].find("http") > -1 ):
-        item['temp'] = True
-    
-    elif ( item['file_original_path'].find("rar://") > -1 ):
-        item['rar']  = True
-        item['file_original_path'] = os.path.dirname(item['file_original_path'][6:])
-    
-    elif ( item['file_original_path'].find("stack://") > -1 ):
-        stackPath = item['file_original_path'].split(" , ")
-        item['file_original_path'] = stackPath[0][8:]
   
     Search(item)  
 elif params['action'] == 'manualsearch':
-    res=re.findall('(.*?)(\d{1,3})x(\d{1,3})', urllib.unquote(params['searchstring']), re.IGNORECASE)
-    if res:
-        item = {}
-        item['tvshow']=res[0][0]
-        lres=len(item['tvshow'])
-        if item['tvshow'][lres-1:lres]==" ":
-            item['tvshow']=item['tvshow'][0:lres-1]
-        item['season']=res[0][1]
-        item['episode']=res[0][2]
-        item['3let_language']=[]
-        for lang in urllib.unquote(params['languages']).decode('utf-8').split(","):
-            item['3let_language'].append(xbmc.convertLanguage(lang,xbmc.ISO_639_2))
-        Search(item) 
+    item = parseSearchString(params['searchstring'])
+    if item:
+        langstring = urllib.unquote(params['languages']).decode('utf-8')
+        item['3let_language'] = [xbmc.convertLanguage(lang,xbmc.ISO_639_2) for lang in langstring.split(",")]
+        Search(item)
     else:
-        notify(__language__(32002))      
+        notify(__language__(32002))
```

Comment:
In the next update please remove all the \__name\__ style labels. We're all guilty of using them, but they are reserved for python internals naming and should not be used. Thanks.